### PR TITLE
[IN PROGRESS] 56 multi region

### PR DESCRIPTION
**Pull request by rustyrazorblade**

### CODE REVIEW

File: src/main/kotlin/com/thelastpickle/tlpcluster/terraform/Configuration.kt [hunk 1]
```diff
@@ -75,19 +78,25 @@ class Configuration(val ticket: String,
...
+
         setVariable("email", email)
         setVariable("key_name", context.userConfig.keyName)
         setVariable("key_path", context.userConfig.sshKeyPath)
         setVariable("region", region)
-        setVariable("zones", Variable(listOf("us-west-2a", "us-west-2b", "us-west-2c"), "list"))
+
+        val azs = regionLookup.getAzs(region)
+
+        setVariable("zones", Variable(azs))
```
Comment (ossarga) [inline src/main/kotlin/com/thelastpickle/tlpcluster/terraform/Configuration.kt]: Do you think we still need the Terraform variables now? We started deleting them when auto creation of the security groups was added. We should probably remove them all them if they are unused. What do you think?

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Init.kt [hunk 2]
```diff
@@ -75,17 +70,17 @@ class Init(val context: Context) : ICommand {
             FileUtils.copyInputStreamToFile(input, output)
         }
 
-        val config = Configuration(ticket, client, purpose, region = context.userConfig.region, context = context)
+        val region = context.userConfig.region
+        val config = Configuration(ticket, client, purpose, region =  context.userConfig.region , context = context)
```
Comment (ossarga) [inline src/main/kotlin/com/thelastpickle/tlpcluster/commands/Init.kt]: Why do we need to assign `region` its own variable on line 73? Why is `region` an optional when we initialise the `Configuration` class? On line 80 it looks like we override whatever region is set to. Consider removing lines 73 and 80, and then change line 74 to never allow for the `region` to be optional:
```
val config = Configuration(ticket, client, purpose, context.userConfig.region , context = context)
```

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Init.kt [hunk 3]
```diff
@@ -75,17 +70,17 @@ class Init(val context: Context) : ICommand {
...
         config.numStressInstances = stressInstances
-        config.monitoring = monitoringEnabled
 
         config.region = region
-        config.stressAMI = "ami-51537029"
-        config.stressInstanceType = instanceType
+
         config.cassandraInstanceType = instanceType
-        config.monitoringInstanceType = instanceType
+//        config.cassandraAMI = ami
```
Comment (ossarga) [inline src/main/kotlin/com/thelastpickle/tlpcluster/commands/Init.kt]: Should we remove this line?

File: src/main/kotlin/com/thelastpickle/tlpcluster/terraform/Configuration.kt [hunk 4]
```diff
@@ -24,17 +27,17 @@ class Configuration(val ticket: String,
...
-    var stressAMI = "ami-51537029"
-    var stressInstanceType = "c5d.2xlarge"
+
+    var stressInstanceType = "c3.2xlarge"
 
     //monitoring
-    var monitoring = false
-    var monitoringAMI = "ami-51537029"
-    var monitoringInstanceType = "c5d.2xlarge"
+    var monitoring = true
```
Comment (ossarga) [inline src/main/kotlin/com/thelastpickle/tlpcluster/terraform/Configuration.kt]: This variable can be removed now that on line 131 we always launch a monitoring instance.

File: src/main/kotlin/com/thelastpickle/tlpcluster/Context.kt [hunk 5]
```diff
@@ -92,7 +94,11 @@ data class Context(val tlpclusterUserDirectory: File) {
          * Used only for testing
          */
         fun testContext() : Context {
-            val testTempDirectory = Files.createTempDirectory("tlpcluster")
+            File("build/testContext")
```
[the remark this answers is not in the document]
Comment (rustyrazorblade) [reply src/main/kotlin/com/thelastpickle/tlpcluster/Context.kt]: Good catch

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Init.kt (hunk 2 above)
Comment (rustyrazorblade) [reply src/main/kotlin/com/thelastpickle/tlpcluster/commands/Init.kt]: You're right.  It was a quick and sloppy change on my end and I didn't realize I could just delete some lines.

File: src/main/kotlin/com/thelastpickle/tlpcluster/commands/Init.kt (hunk 3 above)
Comment (rustyrazorblade) [reply src/main/kotlin/com/thelastpickle/tlpcluster/commands/Init.kt]: Yes

**Comment by ossarga** [comment]
Merged manually in commit: https://github.com/thelastpickle/tlp-cluster/commit/2cb1f2667ecdcedf4b34814c028135ff61dc953b

**Pull request closed without merging**